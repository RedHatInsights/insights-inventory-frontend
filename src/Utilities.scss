// Sass Imports

@import 'Utilities/_variables.scss';
@import 'Utilities/_colors.scss';
@import 'Utilities/_helpers.scss';
@import 'Utilities/_mixins.scss';

@import '~@redhat-cloud-services/frontend-components/index.css';
