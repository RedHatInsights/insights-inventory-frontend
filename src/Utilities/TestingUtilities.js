import React from 'react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';

export const mountWithRouter = (Component, initialEntries) => {
  const wrapper = mount(
    <MemoryRouter initialEntriest={initialEntries}>{Component}</MemoryRouter>
  );

  return wrapper;
};

export const TestWrapper = ({
  children,
  routerProps = { initialEntries: ['/'] },
  path,
  store = {},
}) => {
  return (
    <MemoryRouter {...routerProps}>
      <Provider store={store}>
        {path ? (
          <Routes>
            <Route path={path} element={children} />
          </Routes>
        ) : (
          children
        )}
      </Provider>
    </MemoryRouter>
  );
};

TestWrapper.propTypes = {
  children: PropTypes.any.isRequired,
  routerProps: PropTypes.shape({
    initialEntries: PropTypes.arrayOf(PropTypes.string),
  }),
  path: PropTypes.string,
  store: PropTypes.any,
};
