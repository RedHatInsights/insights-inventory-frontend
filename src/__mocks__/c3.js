// eslint-disable-next-line no-undef
module.exports = () => 'c3';

