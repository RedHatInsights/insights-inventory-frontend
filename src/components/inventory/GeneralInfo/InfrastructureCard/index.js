export { default } from './InfrastructureCard';
export * from './InfrastructureCard';
