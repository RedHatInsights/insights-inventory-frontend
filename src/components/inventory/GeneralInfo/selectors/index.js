export { default } from './selectors';
export * from './selectors';
