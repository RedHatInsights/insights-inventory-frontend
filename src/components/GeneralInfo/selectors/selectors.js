/* eslint-disable camelcase */
import { registered } from '../../../Utilities/index';
import { verifyCollectorStaleness } from '../../../Utilities/sharedFunctions';
function safeParser(toParse, key) {
  try {
    return JSON.parse(toParse);
  } catch {
    return { [key]: toParse };
  }
}

export const propertiesSelector = (
  {
    number_of_cpus,
    number_of_sockets,
    cores_per_socket,
    ramSize,
    disk_devices,
    sap_sids,
    system_purpose,
    cpu_flags,
  } = {},
  { facts } = {}
) => ({
  cpuNumber: number_of_cpus || facts?.rhsm?.CPU_CORES,
  sockets: number_of_sockets || facts?.rhsm?.CPU_SOCKETS,
  coresPerSocket:
    cores_per_socket ||
    (facts?.rhsm?.CPU_CORES &&
      facts?.rhsm?.CPU_CORES &&
      Number(facts?.rhsm?.CPU_CORES, 10) /
        Number(facts?.rhsm?.CPU_SOCKETS, 10)),
  ramSize: ramSize || (facts?.rhsm?.MEMORY && `${facts?.rhsm?.MEMORY} GB`),
  storage:
    disk_devices &&
    disk_devices.map(({ device, label, mount_point, options, type }) => ({
      ...(device && safeParser(device, 'device')),
      label,
      ...(mount_point && safeParser(mount_point, 'mountpoint')),
      ...(options && safeParser(options, 'options')),
      ...(type && safeParser(type, 'mounttype')),
    })),
  sapIds: sap_sids,
  systemPurpose: system_purpose?.usage,
  cpuFlags: cpu_flags,
});

export const operatingSystem = (
  {
    arch,
    operating_system,
    os_kernel_version,
    last_boot_time,
    kernel_modules,
    system_update_method,
  } = {},
  { facts } = {}
) => ({
  release: operating_system,
  kernelRelease: os_kernel_version,
  architecture: arch || facts?.rhsm?.ARCHITECTURE,
  bootTime: last_boot_time,
  kernelModules: kernel_modules,
  systemUpdateMethod: system_update_method,
});

export const biosSelector = ({
  bios_vendor,
  bios_version,
  bios_release_date,
} = {}) => ({
  vendor: bios_vendor,
  version: bios_version,
  releaseDate: bios_release_date,
});

export const bootcSelector = ({ bootc_status } = {}) => ({
  bootedImage: bootc_status?.booted?.image,
  bootedImageDigest: bootc_status?.booted?.image_digest,
  stagedImage: bootc_status?.staged?.image,
  stagedImageDigest: bootc_status?.staged?.image_digest,
  rollbackImage: bootc_status?.rollback?.image,
  rollbackImageDigest: bootc_status?.rollback?.image_digest,
});

export const infrastructureSelector = (
  {
    infrastructure_type,
    infrastructure_vendor,
    public_ipv4_addresses,
    public_dns,
    network = {},
  } = {},
  { facts } = {}
) => ({
  type:
    infrastructure_type ||
    (facts?.rhsm?.IS_VIRTUAL !== undefined &&
      (facts?.rhsm?.IS_VIRTUAL ? 'virtual' : 'physical')) ||
    undefined,
  vendor: infrastructure_vendor,
  public_ipv4_addresses: public_ipv4_addresses,
  ipv4: network.ipv4,
  ipv6: network.ipv6,
  nics: network.interfaces,
  fqdn: public_dns,
});

export const configurationSelector = ({
  installed_packages,
  enabled_services,
  running_processes,
  repositories,
} = {}) => ({
  packages: installed_packages,
  services: enabled_services,
  processes: running_processes,
  repositories,
});

export const collectionInformationSelector = ({
  insights_client_version,
  insights_egg_version,
} = {}) => ({
  client: insights_client_version,
  egg: insights_egg_version,
});

export const getCollectorStatus = (collectorStaleness) => {
  return collectorStaleness
    ? verifyCollectorStaleness(collectorStaleness) !== 'Fresh'
      ? 'Stale'
      : 'Active'
    : 'N/A';
};

export const getDefaultCollectors = (entity) =>
  registered
    ?.filter((reporter) => reporter.label !== 'insights-client not connected')
    .map((reporter) => ({
      name: reporter.label,
      status: getCollectorStatus(
        entity?.per_reporter_staleness[reporter.value]
      ),
      updated: entity?.per_reporter_staleness[reporter.value]?.last_check_in,
      details: {
        name: reporter.idName,
        id: entity?.[reporter.idValue],
      },
    }));

export const systemStatus = ({ stale_timestamp } = {}) => ({
  stale: new Date() > new Date(stale_timestamp),
});
