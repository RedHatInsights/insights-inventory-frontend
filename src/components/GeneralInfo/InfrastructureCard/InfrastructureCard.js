import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import LoadingCard from '../LoadingCard';
import { generalMapper, interfaceMapper } from '../dataMapper';
import { infrastructureSelector } from '../selectors';
import { extraShape } from '../../../constants';

const InfrastructureCardCore = ({
  infrastructure,
  handleClick,
  detailLoaded,
  hasType,
  hasVendor,
  hasPublicIp,
  hasIPv4,
  hasIPv6,
  hasFqdn,
  hasInterfaces,
  extra,
}) => (
  <LoadingCard
    title="Infrastructure"
    isLoading={!detailLoaded}
    cardId="infrastructure-card"
    items={[
      ...(hasType ? [{ title: 'Type', value: infrastructure.type }] : []),
      ...(hasVendor ? [{ title: 'Vendor', value: infrastructure.vendor }] : []),
      ...(hasPublicIp
        ? [
            {
              title: 'Public IP',
              value:
                Array.isArray(infrastructure.public_ipv4_addresses) &&
                infrastructure.public_ipv4_addresses.length > 0
                  ? infrastructure.public_ipv4_addresses[0]
                  : 'Not available',
            },
          ]
        : []),
      ...(hasIPv4
        ? [
            {
              title: 'IPv4 addresses',
              value: infrastructure.ipv4?.length,
              plural: 'addresses',
              singular: 'address',
              target: 'ipv4',
              onClick: () => {
                handleClick(
                  'IPv4',
                  generalMapper(infrastructure.ipv4, 'IP address'),
                );
              },
            },
          ]
        : []),
      ...(hasIPv6
        ? [
            {
              title: 'IPv6 addresses',
              value: infrastructure.ipv6?.length,
              plural: 'addresses',
              singular: 'address',
              target: 'ipv6',
              onClick: () => {
                handleClick(
                  'IPv6',
                  generalMapper(infrastructure.ipv6, 'IP address'),
                );
              },
            },
          ]
        : []),
      ...(hasFqdn
        ? [
            {
              title: 'FQDN',
              value:
                Array.isArray(infrastructure.fqdn) &&
                infrastructure.fqdn.length > 0
                  ? infrastructure.fqdn[0]
                  : 'Not available',
            },
          ]
        : []),
      ...(hasInterfaces
        ? [
            {
              title: 'Interfaces/NICs',
              value: infrastructure.nics?.length,
              singular: 'NIC',
              target: 'interfaces',
              onClick: () => {
                handleClick(
                  'Interfaces/NICs',
                  interfaceMapper(infrastructure.nics),
                  'medium',
                );
              },
            },
          ]
        : []),
      ...extra.map(({ onClick, ...item }) => ({
        ...item,
        ...(onClick && { onClick: (e) => onClick(e, handleClick) }),
      })),
    ]}
  />
);

InfrastructureCardCore.propTypes = {
  detailLoaded: PropTypes.bool,
  handleClick: PropTypes.func,
  infrastructure: PropTypes.shape({
    type: PropTypes.string,
    vendor: PropTypes.string,
    public_ipv4_addresses: PropTypes.arrayOf(PropTypes.string),
    ipv4: PropTypes.array,
    ipv6: PropTypes.array,
    fqdn: PropTypes.arrayOf(PropTypes.string),
    nics: PropTypes.array,
  }),
  hasType: PropTypes.bool,
  hasVendor: PropTypes.bool,
  hasPublicIp: PropTypes.bool,
  hasIPv4: PropTypes.bool,
  hasIPv6: PropTypes.bool,
  hasFqdn: PropTypes.bool,
  hasInterfaces: PropTypes.bool,
  extra: PropTypes.arrayOf(extraShape),
};
InfrastructureCardCore.defaultProps = {
  detailLoaded: false,
  handleClick: () => undefined,
  hasType: true,
  hasVendor: true,
  hasPublicIp: true,
  hasIPv4: true,
  hasIPv6: true,
  hasFqdn: true,
  hasInterfaces: true,
  extra: [],
};

const mapStateToProps = ({
  entityDetails: { entity },
  systemProfileStore: { systemProfile },
}) => ({
  detailLoaded: systemProfile && systemProfile.loaded,
  infrastructure: infrastructureSelector(systemProfile, entity),
});

export const InfrastructureCard = connect(mapStateToProps)(
  InfrastructureCardCore,
);

InfrastructureCard.propTypes = InfrastructureCardCore.propTypes;
InfrastructureCard.defaultProps = InfrastructureCardCore.defaultProps;

export default InfrastructureCard;
