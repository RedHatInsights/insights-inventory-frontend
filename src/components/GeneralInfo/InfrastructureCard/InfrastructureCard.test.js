/* eslint-disable camelcase */
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { useParams } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { RouterWrapper } from '../../../Utilities/TestingUtilities';
import { infraTest, rhsmFacts } from '../../../__mocks__/selectors';
import InfrastructureCard from './InfrastructureCard';
const location = {};

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => location,
  useParams: jest.fn(() => ({ modalId: 'ipv4' })),
}));

describe('InfrastructureCard', () => {
  let initialState;
  let mockStore;

  beforeEach(() => {
    location.pathname = 'localhost:3000/example/path';
    mockStore = configureStore();
    initialState = {
      systemProfileStore: {
        systemProfile: {
          loaded: true,
          ...infraTest,
        },
      },
      entityDetails: {
        entity: {
          facts: {
            rhsm: rhsmFacts,
          },
        },
      },
    };
  });

  it('should render correctly - no data', () => {
    const store = mockStore({ systemProfileStore: {}, entityDetails: {} });
    const view = render(
      <RouterWrapper>
        <InfrastructureCard store={store} />
      </RouterWrapper>
    );
    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should render correctly with data', () => {
    const store = mockStore(initialState);
    const view = render(
      <RouterWrapper>
        <InfrastructureCard store={store} />
      </RouterWrapper>
    );
    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should render correctly with rhsm facts', () => {
    const store = mockStore({
      ...initialState,
      systemProfileStore: {
        systemProfile: {
          loaded: true,
        },
      },
    });
    const view = render(
      <RouterWrapper>
        <InfrastructureCard store={store} />
      </RouterWrapper>
    );
    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should render enabled/disabled', () => {
    const store = mockStore({
      systemProfileStore: {
        systemProfile: {
          loaded: true,
          ...infraTest,
        },
      },
      entityDetails: {
        entity: {},
      },
    });
    const view = render(
      <RouterWrapper>
        <InfrastructureCard store={store} />
      </RouterWrapper>
    );
    expect(view.asFragment()).toMatchSnapshot();
  });

  describe('api', () => {
    it('should call handleClick on ipv4', async () => {
      const store = mockStore(initialState);
      const onClick = jest.fn();
      location.pathname = 'localhost:3000/example/ipv4';
      useParams.mockImplementation(() => ({ modalId: 'ipv4' }));
      render(
        <RouterWrapper>
          <InfrastructureCard handleClick={onClick} store={store} />
        </RouterWrapper>
      );

      await userEvent.click(
        screen.getAllByRole('link', { name: /1 address/i })[0]
      );
      await waitFor(() => {
        expect(onClick).toHaveBeenCalledTimes(1);
      });
    });

    it('should call handleClick on ipv6', async () => {
      const store = mockStore(initialState);
      const onClick = jest.fn();
      location.pathname = 'localhost:3000/example/ipv6';
      useParams.mockImplementation(() => ({ modalId: 'ipv6' }));
      render(
        <RouterWrapper>
          <InfrastructureCard handleClick={onClick} store={store} />
        </RouterWrapper>
      );

      await userEvent.click(
        screen.getAllByRole('link', { name: /1 address/i })[1]
      );
      await waitFor(() => {
        expect(onClick).toHaveBeenCalledTimes(1);
      });
    });

    it('should call handleClick on interfaces', async () => {
      const store = mockStore(initialState);
      const onClick = jest.fn();
      location.pathname = 'localhost:3000/example/interfaces';
      useParams.mockImplementation(() => ({ modalId: 'interfaces' }));
      render(
        <RouterWrapper>
          <InfrastructureCard handleClick={onClick} store={store} />
        </RouterWrapper>
      );

      await userEvent.click(screen.getByRole('link', { name: /1 NIC/i }));
      await waitFor(() => {
        expect(onClick).toHaveBeenCalledTimes(1);
      });
    });
  });

  ['hasType', 'hasVendor', 'hasIPv4', 'hasIPv6', 'hasInterfaces'].map((item) =>
    it(`should not render ${item}`, () => {
      const store = mockStore(initialState);
      const view = render(
        <RouterWrapper>
          <InfrastructureCard store={store} {...{ [item]: false }} />
        </RouterWrapper>
      );

      expect(view.asFragment()).toMatchSnapshot();
    })
  );

  it('should render extra', () => {
    const store = mockStore(initialState);
    const view = render(
      <RouterWrapper>
        <InfrastructureCard
          store={store}
          extra={[
            { title: 'something', value: 'test' },
            {
              title: 'with click',
              value: '1 tests',
              onClick: (_e, handleClick) =>
                handleClick('Something', {}, 'small'),
            },
          ]}
        />
      </RouterWrapper>
    );

    expect(view.asFragment()).toMatchSnapshot();
  });
});
