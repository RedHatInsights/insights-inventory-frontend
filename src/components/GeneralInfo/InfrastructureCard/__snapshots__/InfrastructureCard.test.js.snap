// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfrastructureCard should not render hasIPv4 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-11"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-11"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Infrastructure
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Type
            </dt>
            <dd
              class=""
            >
              test-type
            </dd>
            <dt
              class=""
            >
              Vendor
            </dt>
            <dd
              class=""
            >
              test-vendor
            </dd>
            <dt
              class=""
            >
              IPv6 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv6"
              >
                1 address
              </a>
            </dd>
            <dt
              class=""
            >
              Interfaces/NICs
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//interfaces"
              >
                1 NIC
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`InfrastructureCard should not render hasIPv6 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-12"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-12"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Infrastructure
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Type
            </dt>
            <dd
              class=""
            >
              test-type
            </dd>
            <dt
              class=""
            >
              Vendor
            </dt>
            <dd
              class=""
            >
              test-vendor
            </dd>
            <dt
              class=""
            >
              IPv4 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv4"
              >
                1 address
              </a>
            </dd>
            <dt
              class=""
            >
              Interfaces/NICs
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//interfaces"
              >
                1 NIC
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`InfrastructureCard should not render hasInterfaces 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-13"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-13"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Infrastructure
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Type
            </dt>
            <dd
              class=""
            >
              test-type
            </dd>
            <dt
              class=""
            >
              Vendor
            </dt>
            <dd
              class=""
            >
              test-vendor
            </dd>
            <dt
              class=""
            >
              IPv4 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv4"
              >
                1 address
              </a>
            </dd>
            <dt
              class=""
            >
              IPv6 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv6"
              >
                1 address
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`InfrastructureCard should not render hasType 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-9"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-9"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Infrastructure
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Vendor
            </dt>
            <dd
              class=""
            >
              test-vendor
            </dd>
            <dt
              class=""
            >
              IPv4 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv4"
              >
                1 address
              </a>
            </dd>
            <dt
              class=""
            >
              IPv6 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv6"
              >
                1 address
              </a>
            </dd>
            <dt
              class=""
            >
              Interfaces/NICs
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//interfaces"
              >
                1 NIC
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`InfrastructureCard should not render hasVendor 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-10"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-10"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Infrastructure
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Type
            </dt>
            <dd
              class=""
            >
              test-type
            </dd>
            <dt
              class=""
            >
              IPv4 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv4"
              >
                1 address
              </a>
            </dd>
            <dt
              class=""
            >
              IPv6 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv6"
              >
                1 address
              </a>
            </dd>
            <dt
              class=""
            >
              Interfaces/NICs
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//interfaces"
              >
                1 NIC
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`InfrastructureCard should render correctly - no data 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-1"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-1"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Infrastructure
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Type
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              Vendor
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              IPv4 addresses
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              IPv6 addresses
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              Interfaces/NICs
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`InfrastructureCard should render correctly with data 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-2"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-2"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Infrastructure
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Type
            </dt>
            <dd
              class=""
            >
              test-type
            </dd>
            <dt
              class=""
            >
              Vendor
            </dt>
            <dd
              class=""
            >
              test-vendor
            </dd>
            <dt
              class=""
            >
              IPv4 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv4"
              >
                1 address
              </a>
            </dd>
            <dt
              class=""
            >
              IPv6 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv6"
              >
                1 address
              </a>
            </dd>
            <dt
              class=""
            >
              Interfaces/NICs
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//interfaces"
              >
                1 NIC
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`InfrastructureCard should render correctly with rhsm facts 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-3"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-3"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Infrastructure
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Type
            </dt>
            <dd
              class=""
            >
              virtual
            </dd>
            <dt
              class=""
            >
              Vendor
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
            <dt
              class=""
            >
              IPv4 addresses
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
            <dt
              class=""
            >
              IPv6 addresses
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
            <dt
              class=""
            >
              Interfaces/NICs
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`InfrastructureCard should render enabled/disabled 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-4"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-4"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Infrastructure
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Type
            </dt>
            <dd
              class=""
            >
              test-type
            </dd>
            <dt
              class=""
            >
              Vendor
            </dt>
            <dd
              class=""
            >
              test-vendor
            </dd>
            <dt
              class=""
            >
              IPv4 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv4"
              >
                1 address
              </a>
            </dd>
            <dt
              class=""
            >
              IPv6 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv6"
              >
                1 address
              </a>
            </dd>
            <dt
              class=""
            >
              Interfaces/NICs
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//interfaces"
              >
                1 NIC
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`InfrastructureCard should render extra 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-14"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-14"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Infrastructure
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Type
            </dt>
            <dd
              class=""
            >
              test-type
            </dd>
            <dt
              class=""
            >
              Vendor
            </dt>
            <dd
              class=""
            >
              test-vendor
            </dd>
            <dt
              class=""
            >
              IPv4 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv4"
              >
                1 address
              </a>
            </dd>
            <dt
              class=""
            >
              IPv6 addresses
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//ipv6"
              >
                1 address
              </a>
            </dd>
            <dt
              class=""
            >
              Interfaces/NICs
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//interfaces"
              >
                1 NIC
              </a>
            </dd>
            <dt
              class=""
            >
              something
            </dt>
            <dd
              class=""
            >
              test
            </dd>
            <dt
              class=""
            >
              with click
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//undefined"
              >
                1 tests
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;
